docs/testplan/SmartSwitch-PMON-testplan.md

```diff
@@ -0,0 +1,939 @@
+root-overlay              OK        Filesystem
+var-log                   OK        Filesystem
+routeCheck                OK        Program
+dualtorNeighborCheck      OK        Program
+container_memory_gnmi     OK        Program
+container_eventd          OK        Program
+database:redis            OK        Process
+bgp:zebra                 OK        Process
+bgp:staticd               OK        Process
+teamd:tlm_teamd           OK        Process
+eventd:eventd             OK        Process
+dpu-pdsagent              OK        UserDefine
+dpu-pciemgrd              OK        UserDefine
+dpu-eth_Uplink1/1_status  OK        UserDefine
+dpu-pcie_link             OK        UserDefine
+
+On Switch: “show system-health detail SWITCH"
+Output: Same as “show system-health detail"
+
+On Switch: “show system-health detail all"
+Output: Same as “show system-health detail SWITCH" plus the DPUs output
+
+```
+#### Pass/Fail Criteria
+ *  A proper output should be delayed without any error
+
+### 1.13 Check the show system-health dpu DPUx CLI on the SWITCH
+
+#### Steps
+ * Issue this CLI “show system-health dpu DPU0" on the switch
+ * Verify the DPU0 state information
+ * Check the midplane state, control-plane state, data-plane state
+ * Alter the control-plane state, data-plane of the DPU by causing a container failure and pipeline failure respectively
+ * Verify the state changes get reflected
+ * TBD: midplane state
+ 
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch: “show system-health dpu DPU0"
+
+Name    ID    Oper-Status    State-Detail             State-Value    Time               Reason
+------  ----  -------------  -----------------------  -------------  -----------------  --------------------------------------------------------------------------------------------------------------------------
+DPU0    0     Online         dpu_midplane_link_state  UP             20240605 02:01:13  INTERNAL-MGMT : admin state - UP, oper_state - UP, status - OK, HOST-MGMT : admin state - UP, oper_state - UP, status - OK
+                             dpu_control_plane_state  UP             20240605 02:01:13  All containers are up and running, host-ethlink-status: Uplink1/1 is UP
+                             dpu_data_plane_state     UP             20240605 00:03:55  DPU container named polaris is running, pdsagent running : OK, pciemgrd running : OK
+
+```
+#### Pass/Fail Criteria
+ *  A proper output should be delayed without any error
+
+### 1.14 Check the cold startup of the DPUs and SWITCH
+
+#### Steps
+ * Set the config_db.json such that some DPUs are powered down
+ * Cold start the system with the config
+ * Verify if the expected DPUs  are ON and the remaining are OFF and the  admin_status is set properly
+ 
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+Once the all the containers are up, (give enough time for the DPUs to boot up as well) and the system is stable
+On Switch: “show platform inventory"
```

Comment:
The show platform inventory command should not depend on the DPU state. It should be able to provide the serial number information regardless of the DPU state. The serial number of the DPU is queried using the DPU module API that runs on the switch.

Reply:
Agree, but if some of the basic containers are down this may not work.

Reply:
The "DPU2 -- not present" means if the DPU is powered on or not.  The DPU information from eeprom can't be read if it is powered down.  The test should check the fields only for the DPUs that are powered. Will mention this in the doc.